input/postgres/plans.go

```diff
@@ -25,39 +25,70 @@ FROM
 WHERE
 	plan_type IN ('estimate', 'actual')`
 
+const statPlanSQL string = `
+SELECT
+	userid, dbid, toplevel, queryid, planid,
+	plan, 'estimate', current_timestamp,
+	calls, total_exec_time,
+	0, 0, 0, 0, 0,
+	0, 0, 0, 0,
+	0, 0,
+	0.0, 0.0
+FROM
+	%s.pg_stat_plans(%t)`
+
+const plansExtensionSQL string = `
+SELECT nspname
+  FROM pg_extension pge
+ INNER JOIN pg_namespace pgn ON pge.extnamespace = pgn.oid
+ WHERE pge.extname = 'pg_stat_plans'
+`
+
 // GetPlans collects query execution plans and stats
 func GetPlans(ctx context.Context, c *Collection, db *sql.DB, showtext bool) (state.PostgresPlanMap, state.PostgresPlanStatsMap, error) {
 	var err error
-	var ioTimeFields string
-
-	// Only collect this with Aurora using aurora_stat_plans function for now
-	if !c.PostgresVersion.IsAwsAurora {
-		return nil, nil, nil
-	}
+	var querySql string
 
-	computePlanIdEnabled, err := GetPostgresSetting(ctx, db, "aurora_compute_plan_id")
-	if err != nil {
-		if c.GlobalOpts.TestRun {
-			c.Logger.PrintInfo("Function aurora_stat_plans() is not supported because Aurora version is too old. Upgrade to Aurora PostgreSQL version 14.10, 15.5, or later versions to collect query plans and stats.")
+	if c.PostgresVersion.IsAwsAurora {
+		computePlanIdEnabled, err := GetPostgresSetting(ctx, db, "aurora_compute_plan_id")
+		if err != nil {
+			if c.GlobalOpts.TestRun {
+				c.Logger.PrintInfo("Function aurora_stat_plans() is not supported because Aurora version is too old. Upgrade to Aurora PostgreSQL version 14.10, 15.5, or later versions to collect query plans and stats.")
```

Comment:
I know this is existing code, but is it wise to log that the version is too old when encountering _any_ error here?

Reply:
Yeah, fair point, reworked that a bit.